sgxvm/src/precompiles/compliance_bridge.rs

```diff
@@ -45,119 +47,411 @@ impl LinearCostPrecompileWithQuerier for ComplianceBridge {
+                    Some(address) => Ok(address),
+                    None => Err(()),
+                })
+                .collect();
+
+            let allowed_issuers = match allowed_issuers {
+                Ok(issuers) => issuers,
+                Err(_) => {
+                    return Err(PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: encode(&vec![AbiToken::String(
+                            "one or more invalid issuer addresses".into(),
+                        )]),
+                    });
+                }
+            };
 
             let encoded_request = coder::encode_has_verification_request(
-                user_address.clone().into_address().unwrap(),
-                verification_type.clone().into_uint().unwrap().as_u32(),
-                expiration_timestamp.clone().into_uint().unwrap().as_u32(),
+                user_address,
+                verification_type,
+                expiration_timestamp,
                 allowed_issuers,
             );
 
             match querier::make_request(querier, encoded_request) {
                 Some(result) => {
-                    let has_verification = protobuf::parse_from_bytes::<ffi::QueryHasVerificationResponse>(result.as_slice())
-                        .map_err(|_| PrecompileFailure::Revert {
-                            exit_status: ExitRevert::Reverted,
-                            output: encode(&vec![AbiToken::String("cannot decode protobuf response".into())])
-                        })?;
+                    let has_verification = protobuf::parse_from_bytes::<
+                        ffi::QueryHasVerificationResponse,
+                    >(result.as_slice())
+                    .map_err(|_| PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: encode(&vec![AbiToken::String(
+                            "cannot decode protobuf response".into(),
+                        )]),
+                    })?;
 
-                    let tokens = vec![
-                        AbiToken::Bool(has_verification.hasVerification),
-                    ];
+                    let tokens = vec![AbiToken::Bool(has_verification.hasVerification)];
 
                     let encoded_response = encode(&tokens);
-                    return Ok((ExitSucceed::Returned, encoded_response.to_vec()))
-                },
+                    Ok((ExitSucceed::Returned, encoded_response.to_vec()))
+                }
                 None => {
-                    return Err(PrecompileFailure::Revert {
+                    Err(PrecompileFailure::Revert {
                         exit_status: ExitRevert::Reverted,
-                        output: encode(&vec![AbiToken::String("call to x/compliance failed".into())])
+                        output: encode(&vec![AbiToken::String(
+                            "call to x/compliance failed".into(),
```

Comment:
specify exact error, this string is the same in multiple places, will have a hard time debugging

Reply:
Added the function name to the error string.